src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -4,12 +4,83 @@ public class BinConverter {
 	public static String toString(int x) {
 		// TODO expect x to be in range [-128, 127], return string with 8 binary
 		// digits representing x in 2-complement
-		return "00000000";
+		String result = "";
+		boolean positive = x>=0;
+		if(positive){
+			while(x>0){
+				if(x%2 == 0){
+					result = "0"+result;
+				} else {
+					result = "1"+result;
+				}
+				x = x/2;
+			}
+			int count = 8-result.length();
+			for(int i = 0; i < count; i++ ){
+				result = "0"+result;
+			}
+		} else{
+			x = Math.abs(x);
+			String intermediateResult = "";
+			while(x>0){
+				if(x%2 == 0){
+					intermediateResult = "0"+intermediateResult;
+				} else {
+					intermediateResult = "1"+intermediateResult;
+				}
+				x = x/2;
+			}
+			int count = 8-intermediateResult.length();
+			for(int i = 0; i < count; i++ ){
+				intermediateResult = "0"+intermediateResult;
+			}
+
+			for(int i = 0; i< intermediateResult.length(); i++){
+				if(intermediateResult.charAt(i) == '1'){
+					result += 0;
+				} else{
+					result += 1;
+				}
+			}
+
+
+			for(int i = result.length()-1; i >= 0; i--){
+				int memory = 0;
+				if(result.charAt(i)=='0'){
+					result = replaceChar(result, '1',i);
+					return result;
+				} else {
+					result = replaceChar(result, '0',i);
+				}
+			}
+		}
+		return result;
 	}
 
 	public static int fromString(String text) {
 		// TODO expect text to contain 8 binary digits, parse to int value in
 		// 2-complement
-		return 0;
+		int result = 0;
+		if(text.charAt(0)=='1'){
+			result -= Math.pow(2,7);
```

Comment:
`Math.pow` um 2er-Potenzen als `int` oder `long`-Wert zu berechnen, ist grundsätzlich keine gute Idee, denn das ist eine Fliesskommazahlen-Operation. Das liefert zwar das richtige Resultat, macht aber viel mehr Arbeit als ein einfaches `1 << pow`.
Ansonsten sieht diese Methode ganz ok aus, ausser dass es natürlich noch eleganter ginge, wenn man einfach in jedem Schritt `result = result * 2` rechnet...